evergreen/config.yml

```diff
@@ -1742,23 +1613,23 @@ buildvariants:
   tasks:
   - name: compile_test
 
-- name: macos-1100-arm64-tsan
-  display_name: "MacOS 11 arm64 (TSAN)"
+- name: macos-1100-arm64-release
+  display_name: "MacOS 11 arm64 (Release build)"
   run_on: macos-1100-arm64
   expansions:
     cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.26.3-macos-universal.tar.gz"
     cmake_bindir: "./cmake_binaries/CMake.app/Contents/bin"
     cmake_toolchain_file: "./tools/cmake/xcode.toolchain.cmake"
     cmake_generator: Xcode
     max_jobs: $(sysctl -n hw.logicalcpu)
+    cmake_build_type: Release
     xcode_developer_dir: /Applications/Xcode13.1.app/Contents/Developer
     extra_flags: -DCMAKE_SYSTEM_NAME=Darwin -DCMAKE_OSX_ARCHITECTURES=arm64
-    cmake_build_type: RelWithDebInfo
-    enable_tsan: On
   tasks:
-  # FIXME: tsan is not stable on arm64, fails often with internal errors
-  # - name: compile_test
-  - name: compile_test
+  - name: compile_test_and_package
+  - name: benchmarks
+  - name: long-running-tests
+  - name: swift-build-and-test
```

Comment:
missing 'test-on-exfat' task - it's not run anywhere else

Reply:
to simplify this PR i've undone all the changes to macos in this PR. I've got another PR open to do the xcode 15 upgrade https://github.com/realm/realm-core/pull/7618 and we can discuss any outstanding macos issues there.